lib/webdriveragent.js

```diff
@@ -108,6 +110,18 @@ class WebDriverAgent {
     });
   }
 
+  async checkForDependencies() {
+    if (await fs.hasAccess(this.bootstrapPath + "/Carthage")) {
+      this.xcodebuild.start();
+      return;
+    }
+    let proc = new SubProcess('/bin/bash', ['Scripts/bootstrap.sh'], {cwd: this.bootstrapPath});
```

Comment:
I would say here you would just use `teen_process.exec`:

```js
await exec('/bin/bash', ['Scripts/bootstrap.sh'], {cwd: this.bootstrapPath});
this.xcodebuild.start();
```

another question, what is `xcodebuild.start()` and is it not async?